cocotb_test/simulator.py

```diff
@@ -516,6 +638,8 @@ def get_parameter_commands(self, parameters):
 
     def build_command(self):
 
+        assert self.vhdl_compile_args is None and self.verilog_compile_args is None, "HDL specific compile arguments not implemented for IUS."
```

Comment:
```suggestion
        assert self.vhdl_compile_args is None and self.verilog_compile_args is None, "HDL specific compile arguments not implemented for IUS; use `compile_args`."
```
Though I think it would also work to silently merge and assign them to `compile_args`, that's what the user would do anyway? This way, the arguments could stay the same when switching simulators.

Same applies to Xcelium a bit lower in the file.

Reply:
The problem here is that the user intent is to provide separate compile args for Verilog and VHDL, which is not possible when we only have one compiler call.

One could argue that `compile_args` are very simulator specific, so sharing them between different simulators is probably a bad idea anyway.